Remove heroku and add K8s (#40)
trywilco/wilco-docs: docs/building-your-quest/conditions/k8s-backend-config-var-defined.md

---

> title: k8s_backend_config_var_defined

Reviewer: it's just backend var, users shouldn't care that we're using k8s

Reviewer 2: They do care though. They use the k8s resource and it should be clear that these config vars are k8s related.

Author: Decided to leave as is, since we would like to make users' life easier by guiding them to search k8s for deployment related actions

parent: Conditions
grand_parent: Building Your Quest
---

# k8s_backend_config_var_defined

Category: K8s

> Description: Check if config variable is set (exists) for the K8s backend app.

Reviewer 3: ```suggestion
Description: Check if a config variable is set (exists) for the K8s backend app.
```

Type: Condition

## Description

**Type**: Condition

> Check if config variable is set (exists) for the K8s backend app. If config var exists, it is set on the condition outputs.

Reviewer 3: ```suggestion
Check if a config variable is set (exists) for the K8s backend app. If a config var exists, it is set on the condition outputs.
```

## Params

- **key:** The config var key

## Outputs

> In config var exists, its `value` is set on the conditions outputs.

Reviewer 3: ```suggestion
If a config var exists, its `value` is set on the conditions outputs.
```

It can be referenced by other blocks using `outputs.<condition_name>.value`

## Usage Example

```yaml
if:
  conditions:
  - conditionId: k8s_backend_config_var_defined
    name: new_relic_license_key_config
    params:
      key: NEW_RELIC_LICENSE_KEY
  then: 
    ...
```

> The `k8s_backend_config_var_defined` condition is used to check that user set the `NEW_RELIC_LICENSE_KEY` config var.

Reviewer 3: ```suggestion
The `k8s_backend_config_var_defined` condition is used to check that the user had set the `NEW_RELIC_LICENSE_KEY` config var.
```
